docs/viewcomponents-in-practice.md

```diff
@@ -34,28 +38,66 @@ App-specific ViewComponents translate a domain object (often an ActiveRecord mod
 
 For example: we have a `User::AvatarComponent` that accepts a `User` ActiveRecord object and renders a `Primer::AvatarComponent`.
 
-## Implementation
+## Organization
 
 ### Extract general-purpose ViewComponents
 
 "Good frameworks are extracted, not invented" - [DHH](https://dhh.dk/arc/000416.html)
 
-Just as ViewComponent itself was extracted from GitHub.com, our experience has shown that the best general-purpose components are those extracted from the GitHub application once they've proven useful across more than one area.
+Just as ViewComponent itself was extracted from GitHub.com, our experience has shown that the best general-purpose components are extracted from the GitHub application once they've proven helpful across more than one area.
 
 Our process typically follows the following steps:
 
-1. Single use-case component implemented in application.
-2. Component adapted for general use in multiple locations in application.
+1. Single use-case component implemented in the application.
+2. Component adapted for general use in multiple locations in the application.
 3. Component extracted into [Primer ViewComponents](https://primer.style/view-components/).
 
 ### Reduce permutations
 
-As we build ViewComponents, we should look for opportunities to consolidate similar patterns into a single implementation. We tend to follow typical DRY practices in this regard, such as abstracting once there are three or more similar instances.
+As we build ViewComponents, we should look for opportunities to consolidate similar patterns into a single implementation. We tend to follow standard DRY practices in this regard, such as abstracting once there are three or more similar instances.
+
+### Avoid one-offs
+
+We should aim to minimize the amount of single-use view code that we write. Every time we don't reuse an existing pattern, we create something to keep up to date, increasing the maintenance burden of our applications.
+
+### Expose existing complexity
+
+Refactoring a view to being a ViewComponent often exposes existing complexity. For example, a ViewComponent may need numerous arguments to be rendered, revealing the number of dependencies in the existing view code. This is good! Refactoring to use ViewComponents helps us understand our view code and gives us a foundation for making it better.
+
+## General guidance
+
+### When to use a ViewComponent for an entire route
+
+ViewComponents have less value in single-use cases like replacing a `show` view. However, it can make sense to render an entire route with a ViewComponent when unit testing is valuable, such as for views with many permutations from a state machine.
```

Comment:
I've found it to be easier to convert a page from partials to components from the bottom up. So instead of trying to convert the page up all at once in discussions we extracted the deepest parts of the page into components. e.g. Instead of focusing on `#show` we'd extract a comment form, a comment, reactions, etc. Is that worth documenting here as well?